pandas/tests/series/test_combine_concat.py

```diff
@@ -116,8 +116,32 @@ def test_update(self):
         df = DataFrame([{"a": 1}, {"a": 3, "b": 2}])
         df['c'] = np.nan
 
-        # this will fail as long as series is a sub-class of ndarray
-        # df['c'].update(Series(['foo'],index=[0])) #####
+        df['c'].update(Series(['foo'], index=[0]))
+        expected = DataFrame([[1, np.nan, 'foo'], [3, 2., np.nan]],
+                             columns=['a', 'b', 'c'])
+        assert_frame_equal(df, expected)
+
+    @pytest.mark.parametrize('other_values', [
+        [61, 63],       # int
+        [61., 63.],     # float, but can be cast to int
+        [61.1, 63.1],   # float, cannot be cast to int
+        [(61,), (63,)]  # object
+    ], ids=['int', 'float_castable', 'float', 'object'])
+    @pytest.mark.parametrize('caller_dtype', ['int', 'float', object])
+    def test_update_dtypes(self, caller_dtype, other_values):
+        caller_values = [10, 11, 12]
+        s = Series(caller_values, dtype=caller_dtype)
+        other = Series(other_values, index=[1, 3])
+        s.update(other)
+
+        expected_values = [caller_values[0], other_values[0], caller_values[2]]
+        try:
+            # we keep original dtype whenever possible
+            expected = Series(expected_values, dtype=caller_dtype)
+        except ValueError:
+            expected = Series(expected_values)
```

Comment:
I would consider passing in `expected` as one of the parameters to the test.  I'm kind of iffy on a `try-except` inside a test...

Reply:
Then I couldn't parametrize as I'm doing now - I'd have to use only one parameter over several args, and have 12 different `expected` values to pass. That's neither easy to read nor very extensible, so the try-except (plus comment) isn't such a bad deal, IMO.

Reply:
Okay, that's fair.